src/foam/mlang/mlang.js

```diff
@@ -508,6 +508,11 @@ foam.CLASS({
       swiftCode: 'return true',
       javaCode: 'return true;'
     },
+    {
+      name: 'partialEval',
+      code: function() { return this },
+      javaCode: 'return foam.mlang.MLang.TRUE;'
```

Comment:
Doesn't this already 'return this', which is the same?

Reply:
No, the singleton axiom doesn't work on java.

    foam.mlang.MLang.TRUE == new True();  // false

Reply:
I see. Then that's the real bug.  We need java Singleton support.

Reply:
We can fix this in the future.